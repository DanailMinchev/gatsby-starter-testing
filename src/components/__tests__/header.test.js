import React from "react"
import { render, screen } from "@testing-library/react"

import Header from "../header"

describe(`Header`, () => {
  it(`renders siteTitle`, () => {
    const siteTitle = `Hello World`

    render(<Header siteTitle={siteTitle} />)

    const title = screen.getByText(siteTitle)
    expect(title).toBeInTheDocument()
  })

  it(`renders link`, () => {
    const siteTitle = `Hello World`

    render(<Header siteTitle={siteTitle} />)

    const title = screen.getByText(siteTitle)
    expect(title).toHaveAttribute(`href`, `/`)
  })

  describe(`Snapshots`, () => {
    it(`matches snapshot`, () => {
      const siteTitle = `Hello World`

      const { container } = render(<Header siteTitle={siteTitle} />)

      expect(container).toMatchSnapshot()
    })
  })
})
