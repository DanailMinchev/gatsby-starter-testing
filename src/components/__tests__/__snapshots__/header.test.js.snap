// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshots matches snapshot 1`] = `
<div>
  <header
    style="background: rebeccapurple; margin-bottom: 1.45rem;"
  >
    <div
      style="margin: 0px auto; max-width: 960px; padding: 1.45rem 1.0875rem;"
    >
      <h1
        style="margin: 0px;"
      >
        <a
          href="/"
          style="color: white; text-decoration: none;"
        >
          Hello World
        </a>
      </h1>
    </div>
  </header>
</div>
`;
